### PR TITLE
Fixup ControlNet3D shape issues

### PR DESCRIPTION
**Pull request by JCBrouwer**
- pass num_frames in all the places
- refactor PseudoConv3D to fit Diffusers' style of 3D blocks (B*T,C,H,W shapes and a num_frames arg on .forward())
- fix various shape issues throughout

### CODE REVIEW

File: controlnet3d.py [hunk 1]
```diff
@@ -91,7 +80,7 @@ class ControlNet3DConditioningEmbedding(nn.Module):
     def __init__(
         self,
         conditioning_embedding_channels: int,
-        conditioning_channels: int = 3,
+        conditioning_channels: int = 2,
```
Comment (jfischoff) [inline controlnet3d.py:83]: Does this matter? I thought we end up passing in the right value anyway?

File: train_controlnet3d.py [hunk 2]
```diff
@@ -844,8 +843,10 @@ def load_model_hook(models, input_dir):
         for step, batch in enumerate(train_dataloader):
             with accelerator.accumulate(controlnet):
                 # Convert videos to latent space
-                latents = vae.encode(batch["pixel_values"].to(dtype=weight_dtype)).latent_dist.sample()
-                latents = latents * vae.config.scaling_factor
+                pixel_values = rearrange(batch["pixel_values"], "b f c h w -> (b f) c h w")
+                latents = vae.encode(pixel_values.to(dtype=weight_dtype)).latent_dist.sample()
+                latents *= vae.config.scaling_factor
```
Comment (jfischoff) [inline train_controlnet3d.py:848]: Does this work with training? I could be misremembering, but I thought you can't use *= and += if you want gradients.

File: train_controlnet3d.py [hunk 3]
```diff
@@ -552,7 +550,7 @@ def collate_fn(examples):
     conditioning_pixel_values = torch.stack([example["conditioning_pixel_values"] for example in examples])
     conditioning_pixel_values = conditioning_pixel_values.to(memory_format=torch.contiguous_format).float()
 
-    input_ids = torch.stack([example["input_ids"] for example in examples])
+    input_ids = torch.cat([example["input_ids"] for example in examples])
```
Comment (jfischoff) [inline train_controlnet3d.py:553]: Does this fix the issue I had https://github.com/JCBrouwer/ControlNet3D/compare/main...runners#diff-4bfaa9cca41e6a61f039d45b1b4acd9a9cb6f0385245e7598635649897d2a5efR893

File: train_controlnet3d.py [hunk 4]
```diff
@@ -771,6 +769,7 @@ def load_model_hook(models, input_dir):
     vae.to(accelerator.device, dtype=weight_dtype)
     unet.to(accelerator.device, dtype=weight_dtype)
     text_encoder.to(accelerator.device, dtype=weight_dtype)
+    controlnet.to(accelerator.device, dtype=weight_dtype)
```
Comment (jfischoff) [inline train_controlnet3d.py]: This should happen automatically. I just added it in my branch because I disabled it in prepare but it didn't help so I would not worry about it.

**Review by jfischoff — APPROVED** [review]

File: controlnet3d.py (hunk 1 above)
Comment (JCBrouwer) [reply controlnet3d.py:83]: No it doesn't matter, but it confused me for a moment, so I thought I might as well update it there as well. This still needs to be passed in correctly from the train script.

File: train_controlnet3d.py (hunk 2 above)
Comment (JCBrouwer) [reply train_controlnet3d.py:848]: This part doesn't get any gradients as the encoded latents are the target for the loss function. This is why it's fine to cache all of the latent videos as the VAE isn't generally trained anyway.

File: train_controlnet3d.py (hunk 3 above)
Comment (JCBrouwer) [reply train_controlnet3d.py:553]: I'm not super clear which line the link is referring to, but if it's line 912, then yes. This ensures we don't get an extra unit dimension which we don't need.

File: train_controlnet3d.py (hunk 4 above)
Comment (JCBrouwer) [reply train_controlnet3d.py]: Ah ok, I'll remove it then
